web/src/message_notifications.ts

```diff
@@ -140,29 +165,37 @@ function get_notification_title(message, msg_count) {
     return title_prefix + title_suffix;
 }
 
-export function process_notification(notification) {
+export function process_notification(notification: {
+    message: Message | TestNotificationMessage;
+    desktop_notify: boolean;
+}): void {
     const message = notification.message;
     const content = get_notification_content(message);
     const key = get_notification_key(message);
-    let notification_object;
+    let notification_object: ElectronBridgeNotification | Notification;
     let msg_count = 1;
 
     debug_notification_source_value(message);
 
-    if (desktop_notifications.notice_memory.has(key)) {
-        msg_count = desktop_notifications.notice_memory.get(key).msg_count + 1;
-        notification_object = desktop_notifications.notice_memory.get(key).obj;
+    const notice_memory = desktop_notifications.notice_memory.get(key);
+    if (notice_memory) {
+        msg_count = notice_memory.msg_count + 1;
+        notification_object = notice_memory.obj;
         notification_object.close();
     }
 
     const title = get_notification_title(message, msg_count);
 
     if (notification.desktop_notify) {
-        const icon_url = people.small_avatar_url(message);
+        const icon_url =
+            message.type === "test-notification"
+                ? small_avatar_url_for_test_notification(message)
+                : people.small_avatar_url(message);
+        assert(desktop_notifications.NotificationAPI !== undefined);
```

Comment:
[Apparently](https://bugs.webkit.org/show_bug.cgi?id=247315) `window.Notification` is really `undefined` in iOS Safari private browsing mode. We should probably do `if (notification.desktop_notify && desktop_notifications.NotificationAPI !== undefined)` instead of `assert`.

Reply:
Thanks for calling that out, I've added a commit to do that